Provide translated list of goals to be displayed by the "Goals Capture Step" screen

mashikag opened the pull request:
Creating a hook that will provide a list of translated 'goals' that are to be displayed in the "Capture Goals" step of the setup flow.

#### Changes proposed in this Pull Request

* new `goals` step hook to provide a list of translated goals that are to be displayed in the new `Capture Goals` screen

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

No real testing instructions for this. As long as it does not regress any current functionality, it should introduce no changes to the current UI. 

The hook will be verified at the time of implementing the new `Capture Goals` screen which will consume the returned data.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #63959

mashikag committed:
Provide translated list of goals to be displayed by UI

Creating a hook that will provide a list of translated 'goals' that are to be displayed in the "Capture Goals" step of the setup flow.

yansern commented on client/signup/steps/goals/goals.tsx:
```diff
@@ -0,0 +1,36 @@
+import { GoalId } from './constants';
+import type { Goal } from './types';
+
+export const useIntents = (): Goal[] => {
```
Should this be `useGoals`?

mashikag replied:
Indeed. Thanks for spotting. 👍

yansern commented on client/signup/steps/goals/goals.tsx:
```diff
@@ -0,0 +1,36 @@
+
+	const goals: Goal[] = [
+		{
+			id: GoalId.Write,
+			label: translate( 'Write and publish' ),
```
Since there is prior art in using `key` & `title`, shall we follow the same?
https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/intent-step/intents.tsx#L15-L17

mashikag replied:
Sounds good to me. 👍

yansern commented on client/signup/steps/goals/goals.tsx:
```diff
@@ -0,0 +1,36 @@
+		},
+		{
+			id: GoalId.DIFM,
+			label: translate( 'Hire a professional to design my website' ),
```
This should be marked as `isPremium: true`.

yansern commented on client/signup/steps/goals/constants.ts:
```diff
@@ -0,0 +1,8 @@
+	Import = 'import',
+	Promote = 'promote',
+	Sell = 'sell',
+	Write = 'write',
```
Nit. How about ordering it the way it was laid out on the design, e.g. Write, Promote, Sell, etc.

mashikag replied:
Sure 👍

yansern commented on client/signup/steps/goals/types.ts:
```diff
@@ -0,0 +1,7 @@
+
+export type Goal = {
+	id: GoalId;
+	label: React.ReactChild | string;
```
Is `React.ReactChild` a possible returned type of `useTranslate`?

mashikag replied:
Not of the `useTranslate` but the `translate` function which is returned by `useTranslate`.

yansern replied:
Gotcha.

ivan-ottinger commented:
Just one observation comment: I see we are putting the files inside `client/signup/steps/goals/`. In our other PR (https://github.com/Automattic/wp-calypso/pull/63866) we used `client/signup/steps/goals-step/` (similarly to `intent-step`), but I prefer simpler `goals` directory as is used in this PR. 👍🏼 

→ We will adjust the directory in https://github.com/Automattic/wp-calypso/pull/63866.

mashikag committed:
Apply PR suggestions

- Rename useIntents to useGoals hook
- Re-order GoalKey values
- Use `key` and `title` instead of `id` and `label` as keys of Goal type object

ivan-ottinger commented:
While looking at the file locations again today, there's one thing that's not clear to me.

My understanding was that our new _Goals step_ will live in `client/landing/stepper/declarative-flow/internals/steps-repository/goals/`. But in this PR we are putting the files inside `client/signup/steps/goals` instead. 🤔

mashikag commented:
> While looking at the file locations again today, there's one thing that's not clear to me.
> 
> My understanding was that our new _Goals step_ will live in `client/landing/stepper/declarative-flow/internals/steps-repository/goals/`. But in this PR we are putting the files inside `client/signup/steps/goals` instead. 🤔

You are right @ivan-ottinger ! Let me fix the directory. 👍

mashikag committed:
Moving the 'goals' step directory into 'stepper' project

mashikag committed:
Merge branch 'trunk' into add/onboarding-goals-hook

yansern commented:
Yup, `/client/signup` for `/start`, and `/client/landing/stepper` for all the stepper stuff.

And as to where things should be placed, not too worried at the moment, one could argue the hook could be inside `/client/landing/stepper/hooks`. I think this PR is good to go.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7331366

Hi @mashikag, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Write and publish`
- `Sell goods or products`
- `Promote myself or my business`
- `Hire a professional to design my website`
- `Import my existing website content`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7331366).
